### PR TITLE
Fix up broken dependencies and deprecated methods

### PR DESCRIPTION
**Pull request by lukesteensen**
I ran into a few issues getting everything to compile on master. The fixes were:

1. Follow the reorganization of `tracing-fmt` into `tracing_subscriber`
2. Update all crates to the same version of `'tokio-sync`
3. Stop calling deprecated method on `IndexMap` (should be a change in name only).

### CODE REVIEW

File: tower-balance/Cargo.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ tower-util = "0.1.0"
 slab = "0.4"
 
 [dev-dependencies]
-tracing-fmt = { git = "https://github.com/tokio-rs/tracing.git" }
+tracing-subscriber = { git = "https://github.com/tokio-rs/tracing.git" }
```
Comment (LucioFranco) [inline tower-balance/Cargo.toml]:
```suggestion
tracing-subscriber = "0.1.3"
```

File: tower-balance/src/p2c/service.rs [hunk 2]
```diff
@@ -132,7 +132,7 @@ where
                 .next_ready_index
                 .and_then(|i| Self::repair_index(i, idx, self.ready_services.len()));
             debug_assert!(!self.cancelations.contains_key(key));
-        } else if let Some(cancel) = self.cancelations.remove(key) {
+        } else if let Some(cancel) = self.cancelations.swap_remove(key) {
```
Comment (LucioFranco) [inline tower-balance/src/p2c/service.rs]: does this have the same affect? Why this change?

Comment (lukesteensen) [reply tower-balance/src/p2c/service.rs]: `remove` is deprecated, so our `#![deny(warnings)]` gives a compilation failure. According to the docs, `remove` was the same as `swap_remove`. I imagine they're just clarifying the naming.

Comment (LucioFranco) [reply tower-balance/src/p2c/service.rs]: Yup just checked 👍
